demo/src/App.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { Suspense } from 'react';
+import { Canvas } from '@react-three/fiber';
+import { WindowManager } from '../../src/components/WindowManager';
+import { createWindow } from '../../src/Utils/createWindow';
+import { Text, Image, Container } from '@react-three/uikit';
+import { OrbitControls, Environment } from '@react-three/drei';
+import { TabBarWithText } from './components/main';
+
+const TextWindow: React.FC<{ id: string }> = ({ id }) => (
+  <Container>
+    <Text>This is a text window with ID: {id}</Text>
+  </Container>
+);
+
+const ImageWindow: React.FC = () => (
+  <Container width="100%" height="100%">
+    <Image
+      src="https://techlead.agency/api/ai/images/66a6cff8fa664a0ed1553a62"
+      width={100}
+      height={100}
+      objectFit="cover"
+    />
+  </Container>
+);
+
+const TransparentWindow: React.FC = () => (
+  <Container width="100%" height="100%" padding={20}>
+    <Text fontSize={20} color="white">
+      This is a transparent window
+    </Text>
+  </Container>
+);
+
+const Background = () => {
+  return (
+    <>
+      <Environment preset="forest" background={true} />
```

Comment:
these 1 of components dont seem like a good idea if they are not reused in multiple places
you loose the capability to add things in place, increase the LOC of the file and loose expressivity, because developers familiar with "<Environment/>" can directly read that, while "<Background/>" would be new to them